File: docs/packages/hw1/hw1/src/camera.cpp
#include <iostream>

#include <cstdlib>
#include <ctime>

#include <ros/ros.h>
#include "hw1/object_position.h"


using namespace std;
using namespace ros;

int main (int argc, char **argv){

    int freq = 100;
    int max_size = 1000;

    if (argc >= 2){
        freq = atoi(argv[1]);
        if (freq > 50) freq = 50;
    }
    else if (argc == 3){
        max_size = atoi(argv[2]);
        if (max_size > 100) max_size = 100;
    }

    init(argc,argv,"Camera");

    NodeHandle n;

    Publisher pub = n.advertise<hw1::object_position> ("ObjectPose", max_size);
    cout << "Rate of Publishing = "<< freq << " and Maximum Number of Messages = "<< max_size<<endl;

    Rate rate(freq);

    srand(static_cast<unsigned int>(time(0)));  

    while (ok()){
        hw1::object_position pos;

        pos.xO = static_cast<float>(rand()) / RAND_MAX * 100.0f;
        pos.yO = static_cast<float>(rand()) / RAND_MAX * 100.0f;

        pub.publish(pos);
        rate.sleep();
    }
 
    return 0;
}

File: packages/hw1/hw1/src/Camera.py
import rospy
from hw1.msg import object_position
import numpy as np
import sys

rospy.init_node('Camera')

rate_pub = 100
max_msgs_num = 1000
if len(sys.argv) >= 2:
    rate_pub = int (sys.argv[1])
    if rate_pub > 50:
        rate_pub = 50
elif len(sys.argv) == 3:
    max_msgs_num = int(sys.argv[2])
    if max_msgs_num > 100:
        max_msgs_num = 100

print(f"Rate of Publishing = {rate_pub} and Maximum Number of Messages = {max_msgs_num}")

pub = rospy.Publisher('ObjectPose', object_position ,queue_size=max_msgs_num )

rate = rospy.Rate(rate_pub)

obj_pos = object_position()

while not rospy.is_shutdown():
    obj_pos.xO = 80*np.random.rand()
    obj_pos.yO = 80*np.random.rand()
    pub.publish(obj_pos)
    rate.sleep()